pkgs/applications/networking/mailreaders/mmh/default.nix

```diff
@@ -0,0 +1,22 @@
+{ stdenv, fetchurl, ncurses }:
+stdenv.mkDerivation rec {
+  pname = "mmh";
+  version = "0.4";
+  buildInputs = [ ncurses ];
+  patches = [
+    ./patches/adjust-bash-completion-script.patch
+    ./patches/fix-spelling-errors.patch
+    ./patches/multipart-ascii.patch
```

Comment:
Is there a stable url from which these could be fetched?
Also, please clarify what the "adjust-bash-completion-script.patch" is doing.

Reply:
"Multipart" patch is applied upstream, "bash-completion" and "spelling" are written by me and copied from Debian package.

There is no "stable" location, as in sense of "qmail-patches" (https://tomclegg.ca/qmail/).

Reply:
Also added description for "bash-completion" patch.

Reply:
In nixpkgs we prefer to fetch patches if possible. Please consider upstreaming the bash completion patch. The "patches" subdirectory is not our standard.
This currently doesn't build on darwin, but it's supposed to, so let's mark it as broken there for now.
```diff
diff --git a/pkgs/applications/networking/mailreaders/mmh/default.nix b/pkgs/applications/networking/mailreaders/mmh/default.nix
index 34bdc419aa7..efc74d75491 100644
--- a/pkgs/applications/networking/mailreaders/mmh/default.nix
+++ b/pkgs/applications/networking/mailreaders/mmh/default.nix
@@ -1,12 +1,20 @@
-{ stdenv, fetchurl, ncurses }:
+{ stdenv, fetchurl, fetchpatch, ncurses }:
 stdenv.mkDerivation rec {
   pname = "mmh";
   version = "0.4";
   buildInputs = [ ncurses ];
   patches = [
-    ./patches/adjust-bash-completion-script.patch
-    ./patches/fix-spelling-errors.patch
-    ./patches/multipart-ascii.patch
+    ./adjust-bash-completion-script.patch
+    (fetchpatch {
+      name = "mmh-fix-spelling-errors.patch";
+      url = "http://git.marmaro.de/?p=mmh;a=commitdiff_plain;h=d0855b62a07b36c5c5e9c90adbdbac0489027ddd;hp=b272c556cc930991994028c72ee487b38dada93c";
+      sha256 = "1addimmcgzjkznvnr95qkq2p65s8nj5n6mnrrww6app8zzbad3mr";
+    })
+    (fetchpatch {
+      name = "mmh-multipart-ascii.patch";
+      url = "http://git.marmaro.de/?p=mmh;a=commitdiff_plain;h=774ed3d58a95303e92ebaf9928d7b854d89efc1b;hp=adb62752c1b156c218363d87019a4b93df3b1b6f";
+      sha256 = "10bwka6kn2rpv157y7d8hcr7fwkpf9q4ap6xhxny16iqiqlwikgm";
+    })
   ];
   src = fetchurl {
     url = "http://marmaro.de/prog/mmh/files/mmh-${version}.tar.gz";
@@ -17,6 +25,7 @@ stdenv.mkDerivation rec {
     homepage = "http://marmaro.de/prog/mmh";
     license = licenses.bsd3;
     platforms = platforms.unix;
+    broken = stdenv.isDarwin;
     maintainers = with maintainers; [ kaction ];
   };
 }

```

Reply:
I am okay to drop "bash-completion" patch at all, btw. Is it solution.

BTW, what about packaging not release + patches, but "master" (431604647f89d5aac7b199a7883e98e56e4ccf9e) at time of writing?

Reply:
> I am okay to drop "bash-completion" patch at all, btw. Is it solution?

@KAction I cloned the repo from http://git.marmaro.de/?p=mmh and I noticed some commits are by you. Have you tried suggesting upstream your bash completion patch as well?

> BTW, what about packaging not release + patches, but "master" (431604647f89d5aac7b199a7883e98e56e4ccf9e) at time of writing?

I'm not sure how to judge on this one. However I _can_ testify there are no significant changes between 0.4 and master.

Reply:
> I am okay to drop "bash-completion" patch at all, btw. Is it solution?

Yes. We can keep the patch, but it may then require maintenance.

> BTW, what about packaging not release + patches, but "master" (431604647f89d5aac7b199a7883e98e56e4ccf9e) at time of writing?

That's an option. See https://nixos.org/nixpkgs/manual/#sec-package-naming for guidelines regarding the version string.